game-engine/src/IdrisGen.hs

```diff
@@ -24,17 +24,18 @@ gameToIdris (Game { gameGridSize = gridSize, gameStatus = status }) counts =
   \\n\
   \import ProofSweeperBase\n\
   \\n\
-  \%access public export\n\
   \%default total\n\
   \\n\
+  \public export\n\
   \gridSize : Nat\n\
   \gridSize = " ++ show gridSize ++ "\n\
   \\n\
+  \public export\n\
   \mineNeighboursForSize : Coord -> List Coord\n\
   \mineNeighboursForSize = mineNeighbours gridSize\n\
   \\n\
+  \public export\n\
   \data MineFact : Coord -> MineProp -> Type where\n\
-  \  KnownNotMineIsNotMine : MineFact c (KnownNotMine _) -> MineFact c IsNotMine\n\
```

Comment:
Any reason for removing this one? The NotMineImpliesNonMine causes it to be treated as non-terminating, but this one seems to be okay.

Generally we want as many of the axioms as possible to be constructors and only rely on believe_me when we have to - it makes it easier to verify that the axioms don't lead to unsoundness.

Reply:
Good call. I got excited about moving things around and made this consistent just because it felt like the outlier. I wasn't considering that it's better to keep axioms as constructors.

I've moved it back, and also refactored out another of the `believe_me` uses.